--- vpnc-0.4.0/vpnc.c.cloexec	2007-02-19 21:49:51.000000000 +0100
+++ vpnc-0.4.0/vpnc.c	2007-02-22 10:49:46.000000000 +0100
@@ -2246,6 +2246,8 @@
 			if (s->esp_fd == -1) {
 				error(1, errno, "socket(PF_INET, SOCK_RAW, IPPROTO_ESP)");
 			}
+			fcntl(s->esp_fd, F_SETFD, FD_CLOEXEC);
+
 #ifdef IP_HDRINCL
 			if (setsockopt(s->esp_fd, IPPROTO_IP, IP_HDRINCL, &hincl, sizeof(hincl)) == -1) {
 				error(1, errno, "setsockopt(esp_fd, IPPROTO_IP, IP_HDRINCL, 1)");
