diff -up vpnc-0.4.0/vpnc.c.sizeofbug vpnc-0.4.0/vpnc.c
--- vpnc-0.4.0/vpnc.c.sizeofbug	2007-09-03 17:30:03.000000000 +0200
+++ vpnc-0.4.0/vpnc.c	2007-09-03 17:30:45.000000000 +0200
@@ -1971,7 +1971,7 @@ static void setup_link(struct sa_block *
 		rp->next->next->next = us;
 	}
 
-	gcry_create_nonce((uint8_t *) & msgid, sizeof(&msgid));
+	gcry_create_nonce((uint8_t *) & msgid, sizeof(msgid));
 	if (msgid == 0)
 		msgid = 1;
 
