--- vpnc-0.3.3/vpnc.c.cloexec	2006-11-07 18:00:52.000000000 +0100
+++ vpnc-0.3.3/vpnc.c	2006-11-07 18:06:29.000000000 +0100
@@ -221,6 +221,8 @@
 	if (sock < 0)
 		error(1, errno, "making socket");
 
+	fcntl(sock, F_SETFD, FD_CLOEXEC);
+	
 	/* Give the socket a name. */
 	name.sin_family = AF_INET;
 	name.sin_port = port;
@@ -2216,6 +2218,7 @@
 				if (tunnelfd == -1) {
 					error(1, errno, "socket(SOCK_RAW) failed");
 				}
+				fcntl(tunnelfd, F_SETFD, FD_CLOEXEC);
 			}
 		}
 		if (dh_grp)
